uint32 x_dropzone
uint32 y_dropzone
float32 cAngle